test/run_test.py

```diff
@@ -385,6 +385,7 @@ def run_test(
     extra_unittest_args=None,
     env=None,
 ) -> int:
+    env = env or os.environ.copy()
```

Comment:
Couldn't you just set the desired default instead of `None`: `env=os.environ.copy()`?

Reply:
Previously I'd have linters be unhappy about using mutable objects as default arguments in functions, something about how changes can persist.  I don't know if copying the env would fix the problem, but better safe than sorry